services/payments/secrets.go

```diff
@@ -39,6 +44,44 @@ type ChainAddress struct {
 	Approvals []string `ion:"approvals"`
 }
 
+// Vault represents a key which has been broken into shamir shares and is used for encrypting
+// secrets.
+type Vault struct {
+	PublicKey      string   `ion:"publicKey"`
+	Creator        string   `ion:"creator"`
+	Approvals      []string `ion:"approvals"`
+	Threshold      int      `ion:"threshold"`
+	OperatorKeys   []string `ion:"operatorKeys"`
+	IdempotencyKey string   `ion:"idempotencyKey"`
+	// must be unexported. these values should never be presisted to QLDB
```

Comment:
I'm confused by this comment because it seems like we're storing these when we insert the vault on line 161. Is the goal to never store these encrypted shares in QLDB?

Reply:
Yes, and they won't be in the lines below because of how unexported fields work in Go.

Reply:
Do we have any sort of unit tests for this? This isn't clear that this is how this works from looking at the docs of how unexported fields work.

In this case, datastore.go appears to be in the same package as this, so I'm not seeing how the usage of this language feature would work in this case. It's my first time encountering it though so I'm likely not fully understanding how it works.

Have you checked the QLDB instance manually during testing to make sure they're not being stored?

Reply:
Yes I have checked manually. It's storing without those values, as expected.

Reply:
In follow up discussion the reason this works was because the unmarshall command doesn't exist within this package and therefore when that's called before setting it in QLDB unexported attributes get dropped. This is good.